### PR TITLE
Update for bevy ecs v2

### PR DESCRIPTION
**Pull request by rparrett**
~~A work in progress.~~

~~Things seem to work, but I ran out of time and only fixed one of the examples for now.~~

~~Should be able to finish this up later tonight.~~

This should be good to go.

### CODE REVIEW

**Comment by rparrett** [comment]
Unfortunately, there seems to be an issue and some of the examples aren't working properly at the moment.

sprite - not ok, nothing displayed
breakout - not ok, only score displayed
font_atlas_debug - panics

**Comment by rparrett** [comment]
Won't be able to work on this today, but I tried rebasing against the dev branch (locally) and this didn't seem to help the situation.

**Comment by rparrett** [comment]
Didn't get anywhere this morning, but observed that the `reflected layout` `info!` isn't happening in this PR (but that's true in the examples that **do** work as well)

**Comment by rparrett** [comment]
It looks like I had just falsely remembered the sprite flipping PR getting merged after ecs v2. Ignoring that would certainly cause issues with sprites showing up. Investigating.

**Comment by rparrett** [comment]
This is now rebase against/pointing at the dev branch.

**Comment by rparrett** [comment]
@mrk-its 

Pinging you to let you know that I will likely need to abandon this work due to time constraints.

I've pushed my current efforts to get this working with bevy's new PBR stuff.

This puts this PR in a state where 2d stuff seems to work, but basically everything 3d is broken. It's broken in a way that shaders compile, there are no error messages or warnings, but we just get a grey screen.

I suspect that there's something flying over my head related to texture samplers, or perhaps the camera bindings.

**Comment by rparrett** [comment]
The issue seems related to camera bindings, or bindings in general.

I accidentally found that a simple change in `utils.rs` / `reflect_layout` seems to at least make some stuff show up in 3d_scene. I discovered this while attempting to add a `BindingDescriptor` to the `CameraViewProj` `BindGroupDescriptor` in the same function. (Which got me nothing but webgl errors)

```rust
if name == "CameraPosition" {
    continue;
}
```

**Comment by rparrett** [comment]
This is now working (thanks mockersf) but relies on https://github.com/bevyengine/bevy/pull/1798

**Comment by rparrett** [comment]
There are some lingering questions about whether the manual `BindingDescriptor` overrides in `utils.rs` are actually needed, and a few comments about a couple specific changes here: https://github.com/rparrett/bevy_webgl2/pull/1

File: src/shaders/pbr.frag [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+    output_color.rgb += emissive.rgb * output_color.a;
+
+    // tone_mapping
+    output_color.rgb = reinhard_luminance(output_color.rgb);
+    // Gamma correction.
+    // Not needed with sRGB buffer
+    // output_color.rgb = pow(output_color.rgb, vec3(1.0 / 2.2));
+#endif
+
+    o_Target = output_color;
```
Comment (mrk-its) [inline src/shaders/pbr.frag:381]: I thing we need to add `encodeSRGB` call here, like in other shaders

**Pull request merged**